src/sentry/search/snuba/backend.py

```diff
@@ -49,6 +51,27 @@ def unassigned_filter(unassigned, projects):
     return query
 
 
+def linked_filter(linked, projects):
+    # Note that there is no filter being applied here. We rely on the database's query optimizer to
+    # recognize `WHERE id IN (SELECT group_id FROM sentry_platformexternalissue)` the same as
+    # `JOIN sentry_platformexternalissue ON sentry_platformexternalissue.group_id = id` and then choose
+    # the appropriate indexes.
+    group_ids_with_external_issue_ids = PlatformExternalIssue.objects.values_list(
+        "group_id", flat=True
+    )
```

Comment:
Can someone fact-check me here 😬 ... might be nice to test this query against the production sentry.io database?

An alternative I was considering was to add:
```py
        # To avoid this subquery from returning data across all projects in the database
        group__project_id__in=[p.id for p in projects]
```
inside of a filter.